Make scheduler integration test faster

anfernee opened the pull request:
Not to wait for 30 seconds for every negative test case. This commit
also organizes the test code to make it more readable.

It cuts the test time from 450s to 125s.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #53302

**Special notes for your reviewer**:

**Release note**: 
```release-note
NONE
```

dims commented:
/ok-to-test

k82cn commented:
@anfernee , thanks very much for your work on this PR. But I'd like to know whether every goruntine was stopped as expected before reducing the waiting time. When I built other test, I found the test will wait for a long time if any gorunine not exit :).

Maybe we can run every test separately, and check whether it's closing to the waiting time :).

anfernee commented:
@k82cn this change should be safe in terms of goroutine's concern. This change reduces the test time, because for those cases where no schedule is expected, we shouldn't wait "forever" (30 seconds). What are the other tests? I can help take a look. 

IMO, all sleeps in test is harmful, especially in unit test. In this kind of integration test, sometimes it's hard to avoid that. But still, it'd be great if there is a deterministic way to verify no-schedule, no sleeping. It makes test slow, and it adds randomness. You basically don't know how long to sleep.

resouer commented on test/integration/scheduler/predicates_test.go:
```diff
@@ -612,257 +578,274 @@ func TestInterPodAffinity(t *testing.T) {
 			}
-			err = wait.Poll(time.Second, wait.ForeverTestTimeout, podDeleted(cs, context.ns.Name, test.pod.Name))
+
+			err = wait.Poll(100*time.Millisecond, wait.ForeverTestTimeout, podDeleted(cs, nsName, pod.Name))
```
Make `100*time.Millisecond` global?

anfernee replied:
np.

resouer commented:
Sorry, but the format of release note none is wrong. :) Type a ENTER after ```release-note

bsalamat commented on test/integration/scheduler/predicates_test.go:
```diff
@@ -30,450 +30,424 @@ import (
-	cs := context.clientSet
-	podLabel := map[string]string{"service": "securityscan"}
-	podLabel2 := map[string]string{"security": "S1"}
+var podLabel = map[string]string{"service": "securityscan"}
```
Please keep the labels and the test struct inside the test, as it was before. These are data for that particular test. We may add more tests to the file in the future. Besides, this change is unrelated to the title of this PR and makes the review hard.

anfernee replied:
Understood. Let me cut this PR to the minimal to make this easy to review. Maybe a different PR to  reorg the code.

Commit:
Make scheduler integration test faster

Not to wait for 30 seconds for every negative test case. It cuts the
test time from 450s to 125s.

anfernee commented:
/retest

anfernee commented:
I think the failure is not related to this. will re-trigger later.

bsalamat commented:
/lgtm
/approve no-issue

Thanks, @anfernee!

anfernee commented:
/retest
